server/sonar-server/src/main/java/org/sonar/server/notification/NotificationDaemon.java

```diff
@@ -0,0 +1,137 @@
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.notification;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import org.picocontainer.Startable;
+import org.sonar.api.Properties;
+import org.sonar.api.Property;
+import org.sonar.api.config.Settings;
+import org.sonar.api.notifications.Notification;
+import org.sonar.api.server.ServerSide;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+
+@Properties({
+  @Property(
+    key = NotificationDaemon.PROPERTY_DELAY,
+    defaultValue = "60",
+    name = "Delay of notifications, in seconds",
+    project = false,
+    global = false),
+  @Property(
+    key = NotificationDaemon.PROPERTY_DELAY_BEFORE_REPORTING_STATUS,
+    defaultValue = "600",
+    name = "Delay before reporting notification status, in seconds",
+    project = false,
+    global = false)
+})
+@ServerSide
+public class NotificationDaemon implements Startable {
+  private static final String THREAD_NAME_PREFIX = "sq-notification-service-";
+
+  private static final Logger LOG = Loggers.get(NotificationDaemon.class);
+
+  public static final String PROPERTY_DELAY = "sonar.notifications.delay";
+  public static final String PROPERTY_DELAY_BEFORE_REPORTING_STATUS = "sonar.notifications.runningDelayBeforeReportingStatus";
+
+  private final long delayInSeconds;
+  private final long delayBeforeReportingStatusInSeconds;
+  private final DefaultNotificationManager manager;
+  private final NotificationService service;
+
+  private ScheduledExecutorService executorService;
+  private boolean stopping = false;
+
+  public NotificationDaemon(Settings settings, DefaultNotificationManager manager, NotificationService service) {
+    this.delayInSeconds = settings.getLong(PROPERTY_DELAY);
+    this.delayBeforeReportingStatusInSeconds = settings.getLong(PROPERTY_DELAY_BEFORE_REPORTING_STATUS);
+    this.manager = manager;
+    this.service = service;
+  }
+
+  @Override
+  public void start() {
+    executorService = Executors.newSingleThreadScheduledExecutor(
+      new ThreadFactoryBuilder()
+        .setNameFormat(THREAD_NAME_PREFIX + "%d")
+        .setPriority(Thread.MIN_PRIORITY)
+        .build());
+    executorService.scheduleWithFixedDelay(() -> {
+      try {
+        processQueue();
+      } catch (Exception e) {
+        LOG.error("Error in NotificationService", e);
+      }
+    }, 0, delayInSeconds, TimeUnit.SECONDS);
+    LOG.info("Notification service started (delay {} sec.)", delayInSeconds);
+  }
+
+  @Override
+  public void stop() {
+    try {
+      stopping = true;
+      executorService.shutdown();
+      executorService.awaitTermination(5, TimeUnit.SECONDS);
+    } catch (InterruptedException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either re-interrupt this method or rethrow the "InterruptedException". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2142)